src/web/components/form/__tests__/withClickHandler.test.tsx

```diff
@@ -0,0 +1,56 @@
+/* SPDX-FileCopyrightText: 2024 Greenbone AG
+ *
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import {describe, test, expect, testing} from '@gsa/testing';
+import withClickHandler, {
+  WithClickHandlerProps,
+} from 'web/components/form/withClickHandler';
+import {render, fireEvent} from 'web/utils/Testing';
+
+const TestInput = ({...props}) => <input {...props} type="text" />;
+
+interface TestComponentProps extends WithClickHandlerProps<string> {
+  onChange?: (value: string) => void;
+}
+
+describe('withClickHandler tests', () => {
+  test('should call click handler with value', () => {
+    const Component = withClickHandler<TestComponentProps, string>({
+      valueFunc: (event, props): string => props.value,
```

Comment:
It looks like it's not used also for line 22,37,38
```suggestion
      valueFunc: (_event, props): string => props.value,
```